contracts/rollup.sol

```diff
@@ -361,17 +368,28 @@ contract Rollup is RollupHelpers {
         // start with false state
         bool isDisputeValid = false;
 
+        if (batches[_batch_id - 1].batchType == Types.BatchType.Airdrop) {
+            // Validate Reddit signature and sub reddit approval
+        }
+
         for (uint256 i = 0; i < _txs.length; i++) {
             // call process tx update for every transaction to check if any
             // tx evaluates correctly
-            (newBalanceRoot, fromBalance, toBalance, isTxValid) = processTx(
-                batches[_batch_id - 1].stateRoot,
-                batches[_batch_id - 1].accountRoot,
-                _txs[i],
-                _pda_proof[i],
-                _from_proofs[i],
-                _to_proofs[i]
-            );
+            if (batches[_batch_id - 1].batchType == Types.BatchType.Transfer){
```

Comment:
We match BatchType and for each type, we run their process<Item> respectively.

Reply:
Yep, this is great!